src/sentry/integrations/slack/utils/notifications.py

```diff
@@ -111,40 +115,78 @@ def send_incident_alert_notification(
         if incident.status == IncidentStatus.CRITICAL.value:
             payload["reply_broadcast"] = True
 
-    client = SlackClient(integration_id=integration.id)
     success = False
-    try:
-        response = client.post("/chat.postMessage", data=payload, timeout=5)
-        # response should include a "ts" key that represents the unique identifier for the message
-        # referenced at https://api.slack.com/methods/chat.postMessage#examples
-    except ApiError as e:
-        # Record the error code and details from the exception
-        new_notification_message_object.error_code = e.code
-        new_notification_message_object.error_details = {
-            "url": e.url,
-            "host": e.host,
-            "path": e.path,
-            "data": e.json if e.json else e.text,
-        }
-        logger.info("rule.fail.slack_post", exc_info=True)
+    if features.has("organizations:slack-sdk-metric-alert", organization):
+        try:
+            sdk_client = SlackSdkClient(integration_id=integration.id)
+            sdk_response = sdk_client.chat_postMessage(
+                blocks=json_blocks,
+                text=text,
+                channel=str(channel),
+                thread_ts=thread_ts,
+                reply_broadcast=reply_broadcast,
+            )
+        except SlackApiError as e:
+            # Record the error code and details from the exception
+            new_notification_message_object.error_code = e.response.status_code
+            new_notification_message_object.error_details = {
+                "msg": str(e),
+                "data": e.response.data,
+                "url": e.response.api_url,
+            }
+
+            log_params = {
+                "error": str(e),
+                "incident_id": incident.id,
+                "incident_status": new_status,
+            }
+            logger.info("slack.metric_alert.error", exc_info=True, extra=log_params)
+        else:
+            success = True
+
+            ts = None
+            response_data = sdk_response.data
+            if isinstance(response_data, dict):
+                ts = response_data.get("ts")
```

Comment:
is there not an easier way to grab this from the `sdk_response` object?

Reply:
oh yeah i think i can do response.get("ts")

Reply:
Ahh, I was hoping they had an actual method on the dataclass object due to the SDK! Let me see, it's not a blocker for now though